persistence-modules/hibernate-annotations/src/main/java/com/baeldung/hibernate/customtypes/SalaryType.java

```diff
@@ -73,27 +48,30 @@ public int hashCode(Salary x) {
 
     @Override
     public Salary nullSafeGet(ResultSet rs, int position, SharedSessionContractImplementor session, Object owner) throws SQLException {
-        Salary salary = new Salary();
-        salary.setAmount(rs.getLong(position));
-
+        
         if (rs.wasNull())
             return null;
+        
+        Salary salary = new Salary();
+        
+        String salaryValue = rs.getString(position);
+        
+        salary.setAmount(Long.parseLong(salaryValue.split(" ")[1]));
 
-        salary.setCurrency(rs.getString(position));
+        salary.setCurrency(salaryValue.split(" ")[1]);
 
         return salary;
     }
 
     @Override
     public void nullSafeSet(PreparedStatement st, Salary value, int index, SharedSessionContractImplementor session) throws SQLException {
         if (Objects.isNull(value))
-            st.setNull(index, Types.BIGINT);
+            st.setNull(index, Types.VARCHAR);
         else {
-
-            st.setLong(index, SalaryCurrencyConvertor.convert(
-                    value.getAmount(),
-                    value.getCurrency(), localCurrency));
-            st.setString(index + 1, value.getCurrency());
+            Long salaryValue = SalaryCurrencyConvertor.convert(
```

Comment:
the Salary object values are concatenated into a single String because there is only one column in the database for this type